### DISCUSSION TITLE
add_cyclic_point function in unstructured grid

### DISCUSSION THREAD

**Discussion by jonashuegle** [UXARRAY/uxarray#31] [Q&A]
Is there a way to add an "add_cyclic_point" function with this unstructured grid? I've got the problem with my data, that I'm missing one part of data on the 180 deg axis.

**Comment by erogluorhan** [comment]
Hi, @menaiseen , thanks a lot for asking this! For now, there is no such built-in function, but it sounds like a great idea to add to the [UXarray draft API](https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md). I believe it can be implemented, maybe exploiting cartopy's [add_cyclic_point](https://scitools.org.uk/cartopy/docs/latest/reference/generated/cartopy.util.add_cyclic_point.html) utility, just like we had done in [GeoCAT-viz](https://github.com/NCAR/geocat-viz/blob/main/src/geocat/viz/util/util.py#:~:text=def%20xr_add_cyclic_longitudes(da%2C%20coord)%3A) for structured grids.

@paullric , any thoughts?
